addons/web/static/tests/views/fields/reference_field_tests.js

```diff
@@ -259,6 +259,26 @@ QUnit.module("Fields", (hooks) => {
         );
     });
 
+    QUnit.test("ReferenceField respects no_quick_create", async function (assert) {
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `<form><field name="reference" options="{'no_quick_create': 1}" /></form>`,
+        });
+
+        await editSelect(target, "select", "partner");
+        await click(target, ".o_field_widget[name='reference'] input");
+        await editInput(target, ".o_field_widget[name='reference'] input", "new partner");
+        assert.strictEqual(target.querySelectorAll('.ui-autocomplete .o_m2o_dropdown_option').length, 1,
+            "Dropdown should be opened and have only one item");
+        assert.notEqual(
```

Comment:
this isn't a very robust assertion, as a simple typo is enough to validate it. Instead, assert.strictEqual the content of the available option